### PR TITLE
Persistence fix

### PR DESCRIPTION
**Pull request by alexcjohnson**
Edge case discovered while writing persistence docs: failure when enabling/disabling persistence on an existing component.

As discussed with @byronz I also generalized some of our test methods to accept either an element or a CSS selector.

- [x] I have added entry in the `CHANGELOG.md`

### CODE REVIEW

File: dash-renderer/src/persistence.js [hunk 1]
```diff
@@ -280,38 +280,41 @@ const getValsKey = (id, persistedProp, persistence) =>
...
     const element = Registry.resolve(layout);
-    const persisted_props =
-        props.persisted_props || element.defaultProps.persisted_props;
-    const persistence_type =
-        props.persistence_type || element.defaultProps.persistence_type;
-    if (!persisted_props || !persistence_type) {
-        return {props};
-    }
-    return {id, props, element, persistence, persisted_props, persistence_type};
+    const getVal = prop => props[prop] || (element.defaultProps || {})[prop];
```
Comment (Marc-Andre-Rivet) [inline dash-renderer/src/persistence.js]: `null` and `0` could be valid non-default values

Comment (Marc-Andre-Rivet) [reply dash-renderer/src/persistence.js]: Ah ok, just used for persisted_props / persistence_type

File: .circleci/config.yml [hunk 2]
```diff
@@ -200,10 +199,17 @@ jobs:
...
           at: ~/dash
       - run:
-          name: 🧪 Run Integration Tests
+          name: ️️🏗️  Install packages
           command: |
             . venv/bin/activate && cd packages && ls -la
-            find . -name "*.gz" | xargs pip install --no-cache-dir --ignore-installed && pip list | grep dash && cd ..
+            find . -name "*.gz" | xargs pip install --no-cache-dir --ignore-installed && cd ..
+            sed -i '/dash/d' requires-install.txt
+            pip install --no-cache-dir --ignore-installed .
```
Comment (alexcjohnson) [inline .circleci/config.yml]: FYI @byronz - the issue I was having in this PR is related to the original one that led to the `sed` call during `build-core`: sometimes (seemingly at random) the `dash-renderer` we prepared in `packages` was being overwritten by a published one from PyPI - so I used essentially the same solution, but extended to `dash-renderer` by using the pattern `/dash/d` rather than `/dash-/d`.

Also I skipped creating a tarball for dash itself because (a) we already have everything we need here, dash has no build artifacts, and (b) that made it easier to modify its requirements.

File: tests/integration/renderer/test_persistence.py [hunk 3]
```diff
@@ -354,3 +364,102 @@ def reset_names(n):
...
+    @app.callback(Output('out', 'children'), [Input('persisted2', 'value')])
+    def set_out(val):
+        return val
+
+    dash_duo.start_server(app)
+
+    dash_duo.wait_for_text_to_equal('#out', 'a')
+
+    dash_duo.find_element('#persistence-val').send_keys('s')
+    time.sleep(0.2)
```
Comment (Marc-Andre-Rivet) [inline tests/integration/renderer/test_persistence.py]: I see you've removed all the other `sleep`. Is this one special / still required?

Comment (alexcjohnson) [reply tests/integration/renderer/test_persistence.py]: yes - I'm expecting the same text both before and after the `send_keys`, so there's really nothing to wait for; I inserted an explicit wait just to ensure the keypress had fully registered.
